vm_eval.c

```diff
@@ -2169,35 +2176,55 @@ specific_eval(int argc, const VALUE *argv, VALUE self, int singleton, int kw_spl
 
 /*
  *  call-seq:
- *     obj.instance_eval(string [, filename [, lineno]] )   -> obj
- *     obj.instance_eval {|obj| block }                     -> obj
+ *    instance_eval {|self| ... } -> object
+ *    instance_eval(string, filename = nil, lineno = nil) -> string
+ *
+ *  With a block given, passes +self+ to the block
+ *  and returns the block's return value;
+ *  this gives the block full access to +self+:
```

Comment:
Can we use the term "context" here? It's a term used to describe executing as if inside of another object.

```suggestion
 *  this executes within the context of +self+:
```